hermetic_build/common/model/config_change.py

```diff
@@ -74,16 +74,17 @@ def __init__(
         self.baseline_config = baseline_config
         self.current_config = current_config
 
-    def get_changed_libraries(self) -> Optional[list[str]]:
+    def get_changed_libraries(self) -> list[str]:
         """
         Returns a unique, sorted list of library name of changed libraries.
-        None if there is a repository level change, which means all libraries
-        in the current_config will be generated.
+        A special list [ALL_LIBRARIES], will be returned if there is a
+        repository level change, which means all libraries in the current_config
+        will be generated.
 
         :return: library names of change libraries.
         """
         if ChangeType.REPO_LEVEL_CHANGE in self.change_to_libraries:
-            return ConfigChange.ALL_LIBRARIES_CHANGED
+            return [ConfigChange.ALL_LIBRARIES_CHANGED]
```

Comment:
The previous behavior was that we return a list if there are changed libraries or a String `ConfigChange.ALL_LIBRARIES_CHANGED` if there is no changes?

Reply:
The previous behavior:
- Return `None` if all libraries changed
- String of library names if some libraries changed
- Empty string if no library changed

The problem is `None` is interpreted as empty string in the shell script.